libraries/botbuilder-core/src/memoryStorage.ts

```diff
@@ -51,6 +52,7 @@ export class MemoryStorage  implements Storage {
     }
 
     public write(changes: StoreItems): Promise<void> {
+        if (!changes) { throw new ReferenceError(`Changes are required when writing.`); }
```

Comment:
>if (!changes) { throw new ReferenceError(`Changes are required when writing.`); } [](start = 8, length = 81)

move inside the promise

Reply:
This isn't me debating, so much as trying to learn. Is this a style thing or does it provide a functional benefit?

Reply:
If these methods were coded using async/await then technically everything would be inside the promise but my comment was more targeted at the fact that if you're not using async/await to consume this then you now need to handle both an exception or the promise being rejected.  If its inside the promise you just need to handle the promise rejection.

Reply:
In your example its the outer async function that's mapping the throw into a rejected promise.

Reply:
@stevenic Gotcha. Since the promise is part of a `map` loop, would it be better (or work) to `reject` instead of `throw`? I'm sure the performance impact of a looping `if (!keys)` check is negligible, I'd just like to avoid it, if possible.

Reply:
@Stevenic. Oh wow I feel dumb. Completely disregard all of this. I didn't see that these were on `MemoryStorage` and not on `CosmosDbPartitionedStorage`. Ugh.

Reply:
Done.